assume_role.sh

```diff
@@ -1,83 +1,111 @@
-    unset AWS_ACCESS_KEY_ID
-  fi
-
-  roleArn="arn:aws:iam::"
-  roleArn+="$destinationAccountNumber"
-  roleArn+=":role/"
-  roleArn+="$rolename"
+}
 
-  serialArn="arn:aws:iam::"
-  serialArn+="$sourceAccountNumber"
-  serialArn+=":mfa/"
-  serialArn+="$username"
-
-  commandResult=" "
+get_sts () {
   # allow a blank tokenCode for orgs that don't use an MFA
-  if [ -z "$tokenCode" ]; then
-    commandResult+=$(aws sts assume-role --output json \
-                  --role-arn $roleArn \
-                  --role-session-name iam-role-injector \
-                  --query 'Credentials.[SecretAccessKey, SessionToken, AccessKeyId]')
-  else
-    commandResult+=$(aws sts assume-role --output json \
-                  --role-arn $roleArn \
-                  --role-session-name iam-role-injector \
-                  --serial-number $serialArn \
-                  --query 'Credentials.[SecretAccessKey, SessionToken, AccessKeyId]' \
-                  --token-code $tokenCode)
-  fi
+  echo "Enter MFA token code:"
+  read tokenCode
+
+  case "$scriptArgs" in
+    4)
+      if [ -z "$tokenCode" ]; then
+        read -a commandResult <<< $(aws sts assume-role --output text\
+                      --role-arn $roleArn \
+                      --role-session-name iam-role-injector \
+                      --query 'Credentials.[SecretAccessKey, SessionToken, AccessKeyId]')
+      else
+        read -a commandResult <<< $(aws sts assume-role --output text \
+                      --role-arn $roleArn \
+                      --role-session-name iam-role-injector \
+                      --serial-number $serialArn \
+                      --query 'Credentials.[SecretAccessKey, SessionToken, AccessKeyId]' \
+                      --token-code $tokenCode)
+      fi
+      ;;
+    5)
+      if [ -z "$tokenCode" ]; then
+        read -a commandResult <<< $(aws sts assume-role --output text\
+                      --role-arn $roleArn \
+                      --role-session-name iam-role-injector \
+                      --query 'Credentials.[SecretAccessKey, SessionToken, AccessKeyId]' \
+                      --duration-seconds $durationSeconds)
+      else
+        read -a commandResult <<< $(aws sts assume-role --output text \
+                      --role-arn $roleArn \
+                      --role-session-name iam-role-injector \
+                      --serial-number $serialArn \
+                      --query 'Credentials.[SecretAccessKey, SessionToken, AccessKeyId]' \
+                      --duration-seconds $durationSeconds \
+                      --token-code $tokenCode)
+      fi
+      ;;
+      *)
+        echo "Could not form sts assume-role command! Check arguments."
+        exitCode=1
+  esac
 
   exitCode=$?
+}
 
-  size=${#commandResult}
-  if (( $size > 5 )); then
-    commandResult1=$(echo "$commandResult" | sed '5d' | sed '1d' | tr -d '\040\011\012\015' | sed 's/\"//g')
+set_env_vars () {
+  if (( ${#commandResult[@]} == 3 )); then
```

Comment:
Saving commandResult as an array, with `--output text`, returns exactly three items that don't require stripping via sed/tr.